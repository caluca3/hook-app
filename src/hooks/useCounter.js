import { useState } from "react"
//si mandan counter tomara el counter mandado, si no mandan counter tomara el counter 10 por default
export const useCounter = (initialState=10) => {
    
   const [counter, setCounter] = useState(initialState)
   //No se acostumbra a enviar el factor de las condiciones
    const increment=()=>{
        setCounter(counter+1);
    }
    const decrement=()=>{
        setCounter(counter-1);
    }   
    const reset =()=>{
        setCounter(initialState);
    }
    return {
        counter,
        increment,
        decrement,
        reset
};

}
