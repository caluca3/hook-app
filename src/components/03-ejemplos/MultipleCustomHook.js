import React from 'react';
import { useCounter } from '../../hooks/useCounter';
import { useFetch } from '../../hooks/useFetch';

import './styles.css';

export const MultipleCustomHook = () => {

  const {counter,increment}=useCounter(1);

  const {loading,data}= useFetch(`https://www.breakingbadapi.com/api/quotes/${counter}`);
  //para una condicion logica sencilla  y convetti booleanos
  //se puede realiuzar ejemplo!!data&&data[0]
  // para efectos necesarios
console.log(data);
  const {author,quote} = !!data && data[0];//si la data existe extrae la posicion 0
  
  
 
  return (
    <>
    <div>
        <h1>BreakingbBad quotes</h1>
        <hr/>
        
         {
                loading?(
        <div className='alert alert-info text-center'>
            loading...
        </div>

                ):
        <blockquote className='text-end'> 
            <p className='mb-0'>{quote}</p>
            <br/>
            <footer className='blockquote-footer'>{author}</footer>
        </blockquote>
         }

        {
        !loading 
         &&
 
            ( 
                <button id='btn' className="btn btn-primary"onClick={ increment }>
                    Siguiente quote
                </button>
            )
        }

    </div>
    </>
  );
}
