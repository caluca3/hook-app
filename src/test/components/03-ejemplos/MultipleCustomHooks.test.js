import { renderHook } from '@testing-library/react-hooks';
import React from 'react';
import { MultipleCustomHook } from '../../../components/03-ejemplos/MultipleCustomHook';
import { useFetch } from '../../../hooks/useFetch';

jest.mock( '../../../hooks/useFetch');
describe('pruebas en mnultiples custom hooks', () => { 

    test('Debe mostrarse correctamente', () => { 

        useFetch.mockReturnValue({
            data:null,
            loading:false,
            error:null
        });
        const view = renderHook(()=>MultipleCustomHook())
        expect(view).toMatchSnapshot();
     });
     test('should first', () => { 

        useFetch.mockReturnValue({
            data:[{
                author:'Carlos',
                quote:'Hola mundo'
        }],
            loading:false,
            error:null
        });
        const view = renderHook(()=>MultipleCustomHook());
        console.log(view.rerender());
      })
});
