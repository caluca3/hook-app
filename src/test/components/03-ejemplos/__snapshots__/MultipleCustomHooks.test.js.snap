// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en mnultiples custom hooks Debe mostrarse correctamente 1`] = `
Object {
  "rerender": [Function],
  "result": Object {
    "all": Array [
      <React.Fragment>
        <div>
          <h1>
            BreakingbBad quotes
          </h1>
          <hr />
          <blockquote
            className="text-end"
          >
            <p
              className="mb-0"
            />
            <br />
            <footer
              className="blockquote-footer"
            />
          </blockquote>
          <button
            className="btn btn-primary"
            id="btn"
            onClick={[Function]}
          >
            Siguiente quote
          </button>
        </div>
      </React.Fragment>,
    ],
    "current": <React.Fragment>
      <div>
        <h1>
          BreakingbBad quotes
        </h1>
        <hr />
        <blockquote
          className="text-end"
        >
          <p
            className="mb-0"
          />
          <br />
          <footer
            className="blockquote-footer"
          />
        </blockquote>
        <button
          className="btn btn-primary"
          id="btn"
          onClick={[Function]}
        >
          Siguiente quote
        </button>
      </div>
    </React.Fragment>,
    "error": undefined,
  },
  "unmount": [Function],
  "waitFor": [Function],
  "waitForNextUpdate": [Function],
  "waitForValueToChange": [Function],
}
`;
